modules/widgets/instagram/instagram.php

```diff
@@ -0,0 +1,439 @@
+			//echo '<p><small>' . sprintf( __( 'Having trouble? Try <a href="%s" target="_blank">logging into the correct account</a> on Instagram.com first.', 'wpcom-instagram-widget' ), 'https://instagram.com/accounts/login/' ) . '</small></p>';
+
+
+			$jetpack_blog_id = Jetpack::get_option( 'id' );
+			$response = Client::wpcom_json_api_request_as_user(
+				sprintf( '/sites/%d/external-services', $jetpack_blog_id )
+			);
+			$body = json_decode( $response['body'] );
+			$connect_URL = $body->services->instagram->connect_URL;
+			$url = add_query_arg(
+				array(
+					'siteurl' => site_url() . '/wp-admin/widgets.php',
+					'jetpack' => true,
+					'instagram_widget_id' => $this->number,
+				),
+				$connect_URL
+			);
+			echo '<a class="button-primary" href="' . $url . '">Authorize Instagram Access</a>';
+
+			return;
+		}
+
+		// Connected account
+		$page = ( is_customize_preview() ) ? 'customize.php' : 'widgets.php';
+		$query_args = array(
+			'instagram_widget_id' => $this->number,
+			'instagram_widget'    => 'remove_token',
+			'nonce'               => wp_create_nonce( 'instagram-widget-remove-token-' . $this->number . '-' . $instance['token_id'] ),
+		);
+
+		if ( is_customize_preview() ) {
+			$query_args['autofocus[panel]'] = 'widgets';
+		}
+
+		$remove_token_id_url = add_query_arg( $query_args, admin_url( $page ) );
+/*
+		// A legacy API connection, prompt user to delete connection and reconnect
+		if ( $this->is_legacy_token( $instance['token_id'] ) ) {
+			//$connect_url = add_query_arg( array( 'instagram_widget_id' => $this->number, 'is_customizer' => is_customize_preview() ), wpcom_keyring_get_connect_url( 'instagram_basic_display', 'instagram-widget' ) );
+			//echo '<p>' . __( '<strong>Your current connection will stop working on 30 March 2020 due to changes in Instagram\'s service. <br /><br />Please reconnect to Instagram in order to continue using this widget</strong>', 'wpcom-instagram-widget' ) . '</p>';
+			//echo '<p style="text-align:center"><a class="button-primary" target="_top" href="' . esc_url( $connect_url ) . '">' . __( 'Authorize Instagram Access', 'wpcom-instagram-widget' ) . '</a></p>';
+
+
+			$jetpack_blog_id = Jetpack::get_option( 'id' );
+			$response = Client::wpcom_json_api_request_as_user(
+				sprintf( '/sites/%d/external-services', $jetpack_blog_id )
+			);
+			$body = json_decode( $response['body'] );
+			$connect_URL = $body->services->instagram->connect_URL;
+			$url = add_query_arg(
+				array(
+					'siteurl' => site_url() . '/wp-admin/widgets.php',
+					'jetpack' => true,
+					'instagram_widget_id' => $this->number,
+				),
+				$connect_URL
+			);
+			echo '<a class="button-primary" href="' . $url . '">Authorize Instagram Access</a>';
+
+
+			return;
+		}*/
+
+		echo '<p>' . sprintf( __( '<strong>Authorized Account</strong><br /> <a href="%1$s">%2$s</a> | <a href="%3$s">remove</a>', 'wpcom-instagram-widget' ), $instance['token_id'], $instance['token_id'], $instance['token_id'] ) . '</p>';
```

Comment:
I just replaced everything here with the `token_id` to get this working, but really it should look like the line below. We need to get the `external_name` from Instagram...

Reply:
Done